### PR TITLE
Add workflow instance key as label

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Add workflow instance key as label.

## Motivation and Context

There are use cases only labels could help.

## Have you tested this? If so, how?
 

Unit test.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodResourceTest.java [hunk 1]
```diff
@@ -248,6 +248,13 @@ public void shouldConfigureEnvironmentVariables() {
     assertThat(envVars, hasItem(envVar(DOCKER_ARGS, "hello world")));
     assertThat(envVars, hasItem(envVar(ENVIRONMENT, STYX_ENVIRONMENT)));
     assertThat(envVars, hasItem(envVar(LOGGING, "structured")));
+
+    var annotations = pod.getMetadata().getAnnotations();
+    assertThat(annotations, hasEntry(STYX_WORKFLOW_INSTANCE_ANNOTATION_LABEL, WORKFLOW_INSTANCE.toKey()));
+    assertThat(annotations, hasEntry(DOCKER_TERMINATION_LOGGING_ANNOTATION, "false"));
+
+    var labels = pod.getMetadata().getLabels();
+    assertThat(annotations, hasEntry(STYX_WORKFLOW_INSTANCE_ANNOTATION_LABEL, WORKFLOW_INSTANCE.toKey()));
```
Comment (sonjaer) [inline styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodResourceTest.java:257]: labels?

Comment (honnix) [reply styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodResourceTest.java:257]: Where do you refer?

**Comment by honnix** [comment]
I will take a look at the rest. It's weird it keeps failing.

File: styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodResourceTest.java (hunk 1 above)
Comment (honnix) [reply styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodResourceTest.java:257]: Ah I see. Good catch!

**Comment by honnix** [comment]
Workflow instance is not valid label https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

**Pull request closed without merging**